scripts/traffic_light_detector.py

```diff
@@ -126,25 +156,74 @@ def _image_callback(self, msg: CompressedImage) -> None:
         else:
             self._img_pub.publish(msg)
                 
+    def _reset_count_traffic_light(self) -> None:
+        self._count.blue = 0
+        self._count.red = 0
+        self._count.allowed_time = 0
+    
     def _run(self, _) -> None:
         # initialize when the task type is not traffic light
         if not self._request_flag:
-            self._count.blue = 0
-            self._count.red = 0
+            self._reset_count_traffic_light()
+            self._count.to_start_brightness_judge = 0
+            self._count.no_vehicle_on_crosswalk = 0
+            self._crosswalk_detector.reset_buffer()
+        
         # publish flag if a blue is detected above a threshold value after
         #   a red is detected above a threshold value
         elif self._input_cvimg is not None:
-            signal = self._box_reconginition._judge_signal(input_cvimg=self._input_cvimg, param=self._param, count=self._count)
-
-            if signal == "signal_red":
-                self._count.red += 1
-            elif (
-                signal == "signal_blue"
-                and self._count.red > self._param.count_threshold_red
-            ):
-                self._count.blue += 1
-
-            if self._count.blue > self._param.count_threshold_blue:
+            signal = self._box_recognition._judge_signal(input_cvimg=self._input_cvimg, count=self._count)
+            
+            # traffic light: red -> blue
+            if self._is_blue_detected is True:
+                if signal == "signal_red":
+                    self._count.red += 1
+                elif (
+                    signal == "signal_blue"
+                    and self._count.red > self._param.count_threshold_red_detect_blue
+                ):
+                    self._count.blue += 1
+            # traffic light: blue -> red
+            else:
+                if signal == "signal_blue":
+                    self._count.blue += 1
+                elif (
+                    signal == "signal_red"
+                    and self._count.blue > self._param.count_threshold_blue_detect_red
+                ):
+                    self._count.red += 1
+
+            # Check for crosswalk overlap
+            crosswalk_th_img = self._crosswalk_detector._cumulative_crosswalk(input_cvimg=self._input_cvimg)
+
+            # Check if the vehicle is not on the crosswalk
+            if self._crosswalk_detector._check_overlap_with_crosswalk(input_cvimg=self._input_cvimg, thresholded_img=crosswalk_th_img):
+                self._count.no_vehicle_on_crosswalk += 1
+            else:
+                self._count.no_vehicle_on_crosswalk = 0
+                rospy.logwarn("Vehicle on the crosswalk")
+            
+            if self._is_blue_detected is True:
```

Comment:
```suggestion
            if self._is_detecting_blue is True:
```
青を検出している最中という意味であればこちらの方が良いかも知れません
`_is_blue_detected`は認識済み（現在，青信号）という感じがします

Reply:
なので，`_is_blue_detected`を使うのであればTrue, False逆が読みやすいです

Reply:
`_is_detecting_blue`に修正します